lang/jvm/src/test/scala/com/wavesplatform/lang/ParserTest.scala

```diff
@@ -4,27 +4,30 @@ import com.wavesplatform.lang.Common._
 import com.wavesplatform.lang.v1.Parser
 import com.wavesplatform.lang.v1.Terms.Untyped._
 import com.wavesplatform.lang.v1.Terms._
-import com.wavesplatform.lang.v1.testing.ScriptGen
+import com.wavesplatform.lang.v1.testing.{ScriptGenParser}
 import fastparse.core.Parsed.{Failure, Success}
 import org.scalacheck.Gen
 import org.scalatest.prop.PropertyChecks
 import org.scalatest.{Matchers, PropSpec}
 import scodec.bits.ByteVector
 import scorex.crypto.encode.{Base58 => ScorexBase58}
 
-class ParserTest extends PropSpec with PropertyChecks with Matchers with ScriptGen with NoShrink {
+class ParserTest extends PropSpec with PropertyChecks with Matchers with ScriptGenParser with NoShrink {
 
   def parse(x: String): EXPR = Parser(x).get.value
   def isParsed(x: String): Boolean = Parser(x) match {
     case Success(_, _)    => true
     case Failure(_, _, _) => false
   }
 
+  def withWhitespaces(expr: String): String = whitespaces.sample.get + expr + whitespaces.sample.get
```

Comment:
No need to use `sample`. The good way is 
```
for {
 ws1 <- whitespaces
 ws2 <- whitespaces
} yield ws1 + expr + ws2
```
that will change signature of `toString` to return `Gen[String]`

Reply:
After that, `genElementCheck` will look a little different, you might want to drop `toString` completely

Reply:
hm, parse function works with Strings, so from my point of view we should make transformation to String at the begin, than use Gen[String] through many functions and do final transformation only at the end